packages/amplify-frontend-javascript/lib/frontend-config-creator.js

```diff
@@ -289,13 +289,38 @@ function getCognitoConfig(cognitoResources, projectRegion) {
     federationTarget = 'COGNITO_IDENTITY_POOLS';
   }
 
+  let authData = {};
+
+  if (cognitoResource.frontEndConfig) {
+    if (cognitoResource.frontEndConfig.loginMechanism) {
+      authData.aws_cognito_login_mechanism = cognitoResource.frontEndConfig.loginMechanism;
+    }
+
+    if (cognitoResource.frontEndConfig.signupAttributes) {
+      authData.aws_cognito_signup_attributes = cognitoResource.frontEndConfig.signupAttributes;
+    }
+
+    if (cognitoResource.frontEndConfig.mfaConfiguration) {
+      authData.aws_cognito_mfa_configuration = cognitoResource.frontEndConfig.mfaConfiguration;
+    }
+
+    if (cognitoResource.frontEndConfig.mfaTypes) {
+      authData.aws_cognito_mfa_types = cognitoResource.frontEndConfig.mfaTypes;
+    }
+
+    if (cognitoResource.frontEndConfig.passwordProtectionSettings) {
+      authData.aws_cognito_password_protection_settings = cognitoResource.frontEndConfig.passwordProtectionSettings;
+    }
+  }
+
   return {
     aws_cognito_identity_pool_id: cognitoResource.output.IdentityPoolId,
     aws_cognito_region: projectRegion,
     aws_user_pools_id: cognitoResource.output.UserPoolId,
     aws_user_pools_web_client_id: cognitoResource.output.AppClientIDWeb,
     oauth,
     federationTarget,
+    ...authData,
```

Comment:
If this format was agreed upon during API review then it's fine, but I would consider nesting all of this new config under a field like `frontendAuthConfig`

Reply:
Yes, this what was agreed upon during the API review, I did put the data under frontendAuthConfig block in the amplify-meta.json file